viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
+    user_data["watched"] += [movie]
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] += [movie]
+    return user_data
 
+def watch_movie(user_data, title):
+    for data in user_data["watchlist"]:
+        if data["title"] == title:
+            user_data["watchlist"].remove(data)
+            user_data["watched"].append(data)
+            break
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_ratings = 0 
+    if len(user_data["watched"]) == 0:
+        return 0
+    for watched_movie in user_data["watched"]: 
+        sum_ratings += watched_movie["rating"]
+    return sum_ratings / len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] in genre_dict:
+            genre_dict[movie["genre"]] += 1
+        else:
+            genre_dict[movie["genre"]] = 1
+
+    most_common_genre = None
+    most_common_count = 0 
+
+    for movie_genre, count in genre_dict.items():
+        if count > most_common_count:
+            most_common_genre = movie_genre
+            most_common_count = count
+
+    return most_common_genre
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched_list = user_data["watched"].copy()
+    for friend in user_data["friends"]:
+        for friend_watched_movie in friend["watched"]:
+            if friend_watched_movie in user_watched_list:
+                user_watched_list.remove(friend_watched_movie)
+    return user_watched_list
+
+def get_friends_unique_watched(user_data):
+    movies_watched_by_friend_only_list = []
+    for friend in user_data["friends"]:
+        for friend_watched_movie in friend["watched"]:
+            if friend_watched_movie in movies_watched_by_friend_only_list:
+                continue
+            if friend_watched_movie not in user_data["watched"]:
+                    movies_watched_by_friend_only_list.append(friend_watched_movie)
+    
+    return movies_watched_by_friend_only_list
 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    recommendations = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie in recommendations:
+                continue
+            
+            elif movie in user_data["watched"]:
+                continue
+            
+            elif movie["host"] not in user_data["subscriptions"]:
+                continue
+            
+            else: 
+                recommendations.append(movie)

```

Comment:
Needing to have 3 conditions that use `continue` doesn't really convey the intent of what is going on here to me and impacts this function's readability. A good test of if something is readable is, to ask yourself if you think you could succinctly explain what's happening in a function to someone in a week. For me, I had to use the debugger to understand the logic. Ideally, we'd be able to understand what's happening by just reading. 

Consider reusing the function you already wrote called `get_friends_unique_watched` to get a list of movies only friends have seen. Then you can check for each movie if the value for "host" is in the user's subscriptions list, if so then add the movie to `recommendations`. What would that code look like?